sound/soc/sof/intel/hda.c

```diff
@@ -414,6 +416,22 @@ void hda_dsp_dump_skl(struct snd_sof_dev *sdev, u32 flags)
 	}
 }
 
+/* dump the first 8 dwords representing the extended ROM status */
+static void hda_dsp_dump_ext_rom_status(struct snd_sof_dev *sdev)
+{
+	char msg[HDA_EXT_ROM_STATUS_MSG_SIZE] = "error: extended rom status:";
+	u32 value;
+	int i;
+
+	for (i = 0; i < HDA_EXT_ROM_STATUS_SIZE; i++) {
+		value = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_SRAM_REG_ROM_STATUS + i * 0x4);
+		snprintf(msg + strlen(msg), HDA_EXT_ROM_STATUS_MSG_SIZE - strlen(msg), "0x%x ",
+			 value);
```

Comment:
nitpick: you could use `" 0x%x"` as a format, i.e. with a space before the format. Then you get a space after ':' for free and you don't waste a space in the end. Then also the longest one number can take is " 0x01234567" i.e. 11 characters. You print up to 8 of them, which makes 88. I'm sure without counting, that `strlen(msg)` from the initialisation is a good deal shorter than 256 - 88 - 1 :-) I think what actually I don't find very pretty about this, is that `HDA_EXT_ROM_STATUS_SIZE` and `HDA_EXT_ROM_STATUS_MSG_SIZE` the way they are defined at the top of the file and carry the full namespace nomenclature, appear quite important, whereas they're only used in one function and only for some printing. If they aren't going to be used for more things, how about something like
```
static void hda_dsp_dump_ext_rom_status(struct snd_sof_dev *sdev)
{
	const int rom_status_size = 8;
	const char msg_tmpl[] = "error: extended rom status:";
	char msg[sizeof(msg_tmpl) + rom_status_size * 11];
```
? But yes, just an optional suggestion not to add `#define`s for short-lived values.

Reply:
@ranj063 @lyakh How about Guennadi's proposal but keep definition of HDA_EXT_ROM_STATUS_SIZE (8) at the top. This seems like a more classc magic number that is better not hidden.

Reply:
@lyakh what I dont like is that somebody will have a poblem with the 11 in the size calculation. Its not straightfoward until you altually look at the printed output to see where the 11 comes from. If the problem is that 256 is a much larger size than required, I can reduce it to 112. But yes, in either case, we're going to need the defines for the 8 and the 11 or the 8 and 112.

Reply:
@ranj063 Ack. Maybe better to keep less clever. Many places in kernel just seem use a simple 128 byte stack alloc for debug prints like this. I.e. something the below. This is simpler, in line with rest of code and has only one magic definition.

```
      size_t len = 0;
      char msg[128];
...
        for (i = 0; i < HDA_EXT_ROM_STATUS_SIZE; i++) {
		value = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_SRAM_REG_ROM_STATUS + i * 0x4);
		len += snprintf(msg + len, sizeof(msg) - len, " 0x%x", value);
        }
 
        dev_err(sdev->dev, "error: extended rom status:%s", msg);
```